jax/numpy/lax_numpy.py

```diff
@@ -1488,7 +1489,10 @@ def reduction(a, axis=None, dtype=None, out=None, keepdims=False):
     if out is not None:
       raise ValueError("reduction does not support the `out` argument.")
 
-    a = a if isinstance(a, ndarray) else asarray(a)
+
+    if isinstance(a, list):
+      raise TypeError(f"Reductions don't work on list, pass a ndarray, tuple or scalar instead")
+    a = asarray(a)
```

Comment:
This change appears to be causing `masking_test.py` to fail.

(If nothing else you could kick the problem down the road by duplicating the logic as it was before the change.)

Reply:
Also note you are using an f-string here but aren't formatting anything.

Reply:
On the f-string, the idea was to print a's type, then I removed, but maybe I should put it back, what do you think?
Also, I'm not exactly understanding why does it make masking_test fail, could you help me understand the error?
Thank you very much for the reviews :D

Reply:
> (If nothing else you could kick the problem down the road by duplicating the logic as it was before the change.)
Could you elaborate a little bit on this please?